trieste/acquisition/function.py

```diff
@@ -405,3 +408,93 @@ def probability_of_feasibility(
     mean, var = model.predict(at)
     distr = tfp.distributions.Normal(mean, tf.sqrt(var))
     return distr.cdf(tf.cast(threshold, at.dtype))
+
+
+class MonteCarloAcquisition(SingleModelAcquisitionBuilder):
+    """
+    Builder for the expected improvement function where the "best" value is taken to be the minimum
+    of the posterior mean at observed points.
+    """
+
+    def __init__(self, eps_shape: [int]):
+
+        super().__init__()
+        assert len(eps_shape) == 3
+        self.eps = tf.random.normal(eps_shape, dtype=default_float())  # [S, B, L]
+
+    def predict_f_samples_with_reparametrisation_trick(self, model: ModelInterface, at: QueryPoints) -> tf.Tensor:
+        """
+        Returns samples according to the reparametrization trick. The sample size S is determined by eps,
+        the batch size B must be compatible between eps and at, N is the number of batches of query points
+        and L is the number of latent processes.
+        :param model:
+        :param at: batch of query points [N, D] or [N, B, D]
+        :return: a [S, N, B, L] tensor
+        """
+        # mean, cov = model.predict_f(at, full_cov=True, full_output_cov=True)
+
+        if len(at.shape) == 2:
+            mean, cov = model.predict(at)
+            # mean: [N, L]
+            # cov: [N, L]
+            samples = mean[None, :, None, :] + \
+                      tf.math.sqrt(cov)[None, :, None, :] * self.eps[:, None, ...]  # [S, N, B, L]
+        else:
```

Comment:
why the two branches?

Reply:
it might be good to split these two branches into two functions if we're not going to treat single-point acquisition as batched acquisitions with batch size 1

Reply:
this is very much copying the gpflow style for conditionals, etc.